lib/SIL/IR/SILFunction.cpp

```diff
@@ -159,10 +159,45 @@ SILFunction *SILFunction::create(
     IsDistributed_t isDistributed, IsRuntimeAccessible_t isRuntimeAccessible,
     IsExactSelfClass_t isExactSelfClass, IsThunk_t isThunk,
     SubclassScope classSubclassScope, Inline_t inlineStrategy, EffectsKind E,
-    SILFunction *insertBefore, const SILDebugScope *debugScope) {
+    SILFunction *insertBefore, const SILDebugScope *debugScope, bool isZombie) {
   // Get a StringMapEntry for the function.  As a sop to error cases,
   // allow the name to have an empty string.
   llvm::StringMapEntry<SILFunction*> *entry = nullptr;
+  if (isZombie) {
```

Comment:
I can create a separate helper function for this instead of changing the existing constructor